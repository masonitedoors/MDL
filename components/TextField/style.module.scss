@import "styles/index";

.text-field {
  position: relative;
  display: inline-block;
  width: 100%;
  margin-bottom: $space-s;

  &--with-label {
    margin-top: $space-m;
  }

  /* Input */

  &__input {
    font-family: $font-family;
    font-size: $base-font-size;
    border-radius: 2px;
    border: 1px solid $color-silver;
    color: $color-deepgray;
    box-shadow: inset 0 1px 2px 0 rgba(0, 0, 0, 0.15);
    height: 32px;
    padding: $space-xxs $space-xs;
    box-sizing: border-box;
    transition: box-shadow 200ms ease-in;
    width: 100%;

    &--align-right {
      text-align: right;
    }

    &--align-center {
      text-align: center;
    }

    &--trailing-icon {
      padding-right: $space-l;
    }
  }

  &--light &__input {
    background: white;
  }

  &--dark &__input {
    background: $color-snow;
  }

  &--error &__input {
    border-color: $color-rust;
  }

  &--no-margin:not(.text-field--label-always-above) {
    margin: 0;
  }

  &--no-margin.text-field--label-always-above {
    margin-bottom: 0;
  }

  &--disabled,
  &--readonly {
    opacity: 0.5;
  }

  /* Label */

  &__label {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    left: $space-xs;
    transition: all 200ms;
    pointer-events: none;
    color: $color-slategray;
  }

  &--error &__label {
    color: $color-rust;
  }

  &--active &__label,
  &--label-always-above &__label {
    top: -$space-xs;
    font-size: 10px;
    color: $color-deepgray;
  }

  /* Helper */

  &__helper {
    position: absolute;
    left: $space-xs;
    bottom: -$space-s;
    font-size: 10px;
  }

  &--error &__helper {
    color: $color-rust;
  }

  /* Icon */

  &__icon {
    @include square($size-lg);

    position: absolute;
    top: 50%;
    right: $space-xs;
    opacity: 0;
    transition: opacity 200ms ease-in 100ms;
    transform: translateY(-50%);
    pointer-events: none;
    fill: $color-limerick;

    &--trailing {
      opacity: 1;
    }
  }

  &--error &__label {
    color: $color-rust;
  }

  &--error &__icon {
    opacity: 1;
    fill: $color-rust;
  }
}
