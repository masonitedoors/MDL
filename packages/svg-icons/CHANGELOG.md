# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unrelease] - YYYY-MM-DD
## [1.12.4] - 2021-10-20
### Changed

- Added <title> to new icons.

## [1.12.2] - 2021-09-29
### Added

- Hammer icon
- MVP Dealer icon
- Showroom icon

## [1.12.0] - 2021-02-18

### Changed

- Thumbs Up icon appearance.

## [1.11.1] - 2021-01-15

### Fixed

- Size issues.

## [1.11.0] - 2021-01-15

### Added

- A ton of solid variants have been added.

## [1.10.0] - 2020-10-14

### Added

- Icons for glass frame, glass lite, grid profile and slab texture.
- Icons for social media.

### Changed

- Icons for width, height and thickness.

## [1.9.0] - 2020-09-24

### Added

- Icons for collection, product category, core, customer, percent-vs-number, stile.
- Second icon for unit configuration.

### Changed

- Icon for product type.

## [1.8.0] - 2020-09-15

### Added

- Pricing icons for plant, distributor location and product category.
- Arrow icons for table sorting.

## [1.7.0] - 2020-08-17

### Added

- Pricing icons for door style, height, thickness and width

## [1.6.0] - 2020-07-14

### Added

- Stile and rail icons for panel profile, louver type and sticking

## [1.5.0] - 2020-06-08

### Added

- Lock and unlock icon revisions

## [1.4.0] - 2020-05-13

### Added

- File manager home and sidebar icons

## [1.3.0] - 2020-05-11

### Added

- Various icons.
