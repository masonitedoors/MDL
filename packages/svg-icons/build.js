const fs = require('fs')

const svgFiles = fs
  .readdirSync('./../../vendor/micons/svg/SVG')
  .filter(file => file.match(/\.svg$/))

try {
  fs.mkdirSync('./src/svg')
} catch (e) {
  // ...
}

const exportStatements = svgFiles
  .map(iconFilename => {
    const hyphenatedName = iconFilename.slice(0, -4)
    const camelcaseName = hyphenatedName.replace(/-([A-Za-z0-9])/g, g => (g.match(/[a-z]/) ? g[1].toUpperCase() : g[1]))
    const exportName = `${camelcaseName[0].toUpperCase()}${camelcaseName.slice(1)}`

    const inlineSvgExport = `export { default as m${exportName} } from './../../../vendor/micons/svg/SVG/${hyphenatedName}.svg'`
    const craExport = `export { default as M${exportName} } from './../../../vendor/micons/svg/SVG/${hyphenatedName}.svg'`

    return [inlineSvgExport, craExport].join('\n')
  })
  .join('\n')

fs.writeFileSync('src/index.js', exportStatements)
