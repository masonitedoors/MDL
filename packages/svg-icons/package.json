{
  "name": "@masonite/svg-icons",
  "version": "1.7.0",
  "description": "",
  "main": "dist/index.js",
  "scripts": {
    "build": "npm i && cross-env NODE_PATH=./../../ node build.js && cross-env NODE_PATH=./../../ webpack --mode production"
  },
  "sideEffects": false,
  "keywords": [],
  "publishConfig": {
    "access": "public"
  },
  "author": "Tarik Hamilton",
  "license": "MIT",
  "devDependencies": {
    "@babel/preset-env": "^7.10.4",
    "@babel/preset-react": "^7.10.4",
    "babel-loader": "^8.1.0",
    "svg-inline-loader": "^0.8.2",
    "webpack": "^4.43.0",
    "webpack-cli": "^3.3.12"
  },
  "dependencies": {
    "copy-webpack-plugin": "^4.6.0"
  }
}
