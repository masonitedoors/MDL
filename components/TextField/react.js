import React, { memo, useState } from 'react'
import classNames from 'classnames/bind'
import { mAlertTriangle } from '@masonite/svg-icons'
import PropTypes from 'prop-types'
import style from './style.module.scss'

const cx = classNames.bind(style)

const TextField = ({
  error,
  helper,
  label,
  labelAlwaysAbove,
  onBlur,
  onClick,
  onFocus,
  onChange,
  onKeyDown,
  placeholder,
  trailingIcon,
  type,
  value,
  variant,
  removeMargin,
  align,
  disabled,
  readonly,
}) => {
  const [isActive, setActive] = useState(false)
  const Label = label ? 'label' : 'div'
  const fieldValue = (value || '').toString()

  const TrailingIcon = () => (
    <div
      className={cx(['text-field__icon', trailingIcon && !error && 'text-field__icon--trailing'])}
      dangerouslySetInnerHTML={{ __html: error ? mAlertTriangle : trailingIcon }}
    />
  )

  return (
    <Label
      className={cx('text-field', {
        'text-field--with-label': label,
        'text-field--light': variant === 'light',
        'text-field--dark': variant === 'dark',
        'text-field--error': error,
        'text-field--active': isActive || fieldValue.length,
        'text-field--label-always-above': labelAlwaysAbove,
        'text-field--no-margin': removeMargin,
        'text-field--disabled': disabled,
        'text-field--readonly': readonly,
      })}
    >
      <div className={cx('text-field__label', {
        'text-field--error': error,
      })}>{label}</div>
      <div className={cx('text-field__helper')}>{helper}</div>
      <input
        className={cx('text-field__input', {
          'text-field__input--align-right': align === 'right',
          'text-field__input--align-center': align === 'center',
          'text-field__input--trailing-icon': trailingIcon,
        })}
        type={type}
        placeholder={labelAlwaysAbove || !label ? placeholder : ''}
        onKeyDown={onKeyDown}
        onBlur={event => {
          setActive(false)
          onBlur && onBlur(event.target.value, event)
        }}
        onFocus={event => {
          setActive(true)
          onFocus && onFocus(event.target.value, event)
        }}
        onChange={event => onChange(event.target.value, event)}
        onClick={onClick}
        value={fieldValue}
        disabled={disabled}
        readOnly={readonly}
      />
      <TrailingIcon />
    </Label>
  )
}

export default memo(TextField)

TextField.propTypes = {
  onBlur: PropTypes.func,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func,
  onClick: PropTypes.func,
  error: PropTypes.bool,
  helper: PropTypes.string,
  label: PropTypes.string,
  labelAlwaysAbove: PropTypes.bool,
  onKeyDown: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  trailingIcon: PropTypes.string,
  type: PropTypes.oneOf(['text', 'number', 'email']),
  variant: PropTypes.oneOf(['dark', 'light']),
  removeMargin: PropTypes.bool,
  align: PropTypes.oneOf(['left', 'center', 'right']),
  disabled: PropTypes.bool,
  readonly: PropTypes.bool,
}

TextField.defaultProps = {
  onBlur: undefined,
  onClick: undefined,
  onFocus: undefined,
  error: false,
  helper: '',
  label: undefined,
  labelAlwaysAbove: undefined,
  onKeyDown: undefined,
  placeholder: undefined,
  trailingIcon: undefined,
  type: 'text',
  variant: 'dark',
  removeMargin: false,
  align: 'left',
  value: '',
  disabled: false,
  readonly: false,
}
