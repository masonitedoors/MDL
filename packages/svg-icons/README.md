# @masonite/svg-icons

## Install

```shell
npm install --save @masonite/svg-icons
```

## Usage

If you are using a Create React App application

```js
import { ReactComponent as MBifold } from '@masonite/svg-icons/dist/svg/bifold.svg'
```

or you can import the SVG as a string and implement per your use case or preference.

```js
import { mBifold } from '@masonite/svg-icons'
```

## Development

To develop, clone the `Masonitedoors/MDL` repo with `git clone --recurse-submodules -j8` to get it. Make sure MIcons was cloned to `vendors` in its root. You cannot build without it.

## Deployment

1. Make sure you have done what is mentioned in development.
1. Update the `CHANGELOG.md` for the version you're releasing. If there is an `[Unreleased]` block, change that to the next version following semver.
1. Build the package `npm run build-package:svg-icons` from the monorepo root, followed by `lerna publish`.
