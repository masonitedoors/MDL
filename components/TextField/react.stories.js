import React from 'react'
import { storiesOf } from '@storybook/react'
import { mPhone } from '@masonite/svg-icons'
import {
  withKnobs, boolean, select, text,
} from '@storybook/addon-knobs'
import { withState } from '@dump247/storybook-state'
import TextField from './react'
import Readme from './react.README.md'

storiesOf('TextField', module)
  .addParameters({ readme: { sidebar: Readme } })
  .addDecorator(withKnobs)
  .add(
    'default',
    withState({ value: '' })(({ store }) => {
      const variant = select('Variant', { Light: 'light', Dark: 'dark' }, 'dark')
      const align = select('Align', { Left: 'left', Right: 'right', Center: 'center' }, 'left')
      const label = text('Label', 'Label')
      const labelAlwaysAbove = boolean('Label always above', false)
      const placeholder = text('Placeholder', 'Placeholder')
      const error = boolean('Error State', false)
      const helper = error ? 'Something is wrong.' : ''
      const showTrailingIcon = boolean('Show Trailing Icon', true)
      const disabled = boolean('Disabled', false)
      const readonly = boolean('Readonly', false)
      const removeMargin = boolean('Remove margins', false)
      const trailingIcon = showTrailingIcon && mPhone

      return (
        <TextField
          {...{
            align,
            disabled,
            error,
            helper,
            label,
            labelAlwaysAbove,
            placeholder,
            readonly,
            removeMargin,
            trailingIcon,
            variant,
          }}
          onBlur={value => {}}
          onChange={value => store.set({ value })}
          value={store.state.value}
        />
      )
    }),
  )
  .add(
    'stacked',
    withState({ value: '' })(({ store }) => {
      const variant = select('Variant', { Light: 'light', Dark: 'dark' }, 'dark')
      const align = select('Align', { Left: 'left', Right: 'right', Center: 'center' }, 'left')
      const label = text('Label', 'Label')
      const labelAlwaysAbove = boolean('Label always above', false)
      const placeholder = text('Placeholder', 'Placeholder')
      const error = boolean('Error State', false)
      const helper = error ? 'Something is wrong.' : ''
      const showTrailingIcon = boolean('Show Trailing Icon', true)
      const disabled = boolean('Disabled', false)
      const readonly = boolean('Readonly', false)
      const removeMargin = boolean('Remove margins', false)
      const trailingIcon = showTrailingIcon && mPhone

      return (
        <>
          <TextField
            {...{
              align,
              disabled,
              error: true,
              helper,
              label,
              labelAlwaysAbove,
              placeholder,
              readonly,
              removeMargin,
              trailingIcon,
              variant,
            }}
            onBlur={value1 => {}}
            onChange={value1 => store.set({ value1 })}
            value={store.state.value1}
          />
          <TextField
            {...{
              align,
              disabled,
              error,
              helper,
              label,
              labelAlwaysAbove,
              placeholder,
              readonly,
              removeMargin,
              trailingIcon,
              variant,
            }}
            onBlur={value2 => {}}
            onChange={value2 => store.set({ value2 })}
            value={store.state.value2}
          />
          <TextField
            {...{
              align,
              disabled,
              error,
              helper,
              label,
              labelAlwaysAbove,
              placeholder,
              readonly,
              removeMargin,
              trailingIcon,
              variant,
            }}
            onBlur={value3 => {}}
            onChange={value3 => store.set({ value3 })}
            value={store.state.value3}
          />
        </>
      )
    }),
  )
