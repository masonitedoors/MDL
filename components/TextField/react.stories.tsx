import React, { useState } from "react";
import { Story, Meta } from "@storybook/react";
import { mPhone } from "@masonite/svg-icons";
import TextField from "./react";

const Template = (args) => {
  const [value, setValue] = useState('');

  return (
    <TextField {...args} onChange={(value) => setValue(value)} value={value} />
  );
};

export default {
  title: "TextField",
  component: TextField,
} as Meta;

export const Primary = Template.bind({});

Primary.args = {
  align: "left",
  disabled: false,
  error: false,
  helper: "",
  label: "Phone",
  labelAlwaysAbove: false,
  placeholder: "",
  readonly: false,
  removeMargin: false,
  trailingIcon: mPhone,
  variant: "primary",
};
