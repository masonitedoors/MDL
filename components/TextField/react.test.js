import React from 'react'
import { mount } from 'enzyme'
import { mAlertTriangle } from '@masonite/svg-icons'
import TextField from './react'

describe('TextField', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      value: '',
    }

    wrapper = mount(<TextField {...props} />)
  })

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have an active class when the input is focused', () => {
    expect(
      wrapper
        .find('div')
        .at(0)
        .hasClass('text-field--active'),
    ).toBe(false)
    wrapper
      .find('input')
      .props()
      .onFocus({ target: { value: '' } })
    wrapper.update()
    expect(
      wrapper
        .find('div')
        .at(0)
        .hasClass('text-field--active'),
    ).toBe(true)
  })

  it('should not contain the no-margin modifier when the component initially loads, as the "removeMargin" prop is defaulted to false', () => {
    expect(wrapper.find('.text-field--no-margin').length).toBe(0)
  })

  it('should be wrapped in a <Label> tag if a label is provided', () => {
    expect(wrapper.findWhere(n => n.name() === 'label').length).toBe(0)
    wrapper.setProps({ label: 'TextField Label' })
    expect(wrapper.findWhere(n => n.name() === 'label').length).toBe(1)
  })

  it('should invoke onBlur prop with the value and event when onBlur is invoked', () => {
    const mockEvent = { target: { value: '' } }
    wrapper
      .find('input')
      .props()
      .onBlur(mockEvent)
    expect(props.onBlur).toHaveBeenCalledTimes(1)
    expect(props.onBlur).toHaveBeenCalledWith('', mockEvent)
  })

  it('should invoke onFocus prop with the value and event when onFocus is invoked', () => {
    const mockEvent = { target: { value: '' } }
    wrapper
      .find('input')
      .props()
      .onFocus(mockEvent)
    expect(props.onFocus).toHaveBeenCalledTimes(1)
    expect(props.onFocus).toHaveBeenCalledWith('', mockEvent)
  })

  it('should invoke onChange prop with the value and event when onChange is invoked', () => {
    const mockEvent = { target: { value: 'Text' } }
    wrapper
      .find('input')
      .props()
      .onChange(mockEvent)
    expect(props.onChange).toHaveBeenCalledTimes(1)
    expect(props.onChange).toHaveBeenCalledWith('Text', mockEvent)
  })

  it('should render a trailing icon when trailingIcon has a value', () => {
    const mockSvgString = '<svg></svg>'
    expect(
      wrapper
        .findWhere(n => n.name() === 'TrailingIcon')
        .find('div')
        .props().dangerouslySetInnerHTML.__html,
    ).toBe(undefined)
    wrapper.setProps({ trailingIcon: mockSvgString })
    expect(
      wrapper
        .findWhere(n => n.name() === 'TrailingIcon')
        .find('div')
        .props().dangerouslySetInnerHTML.__html,
    ).toBe(mockSvgString)
  })

  it('should replace the trailingIcon with the error icon when the error prop is true', () => {
    const mockSvgString = '<svg></svg>'
    wrapper.setProps({ trailingIcon: mockSvgString, error: true })
    expect(
      wrapper
        .findWhere(n => n.name() === 'TrailingIcon')
        .find('div')
        .props().dangerouslySetInnerHTML.__html,
    ).toBe(mAlertTriangle)
  })

  it('Should set the no-margin modifier when the "removeMargin" prop is passed in as true', () => {
    wrapper.setProps({ removeMargin: true })
    expect(wrapper.find('.text-field--no-margin').length).toBe(1)
  })

  describe('disabled prop', () => {
    it('Should set the disabled modifier when true', () => {
      wrapper.setProps({ disabled: true })
      expect(wrapper.find('.text-field--disabled').length).toBe(1)
    })
    it('Should not set the disabled modifier when false', () => {
      wrapper.setProps({ disabled: false })
      expect(wrapper.find('.text-field--disabled').length).toBe(0)
    })
  })

  describe('readonly prop', () => {
    it('Should set the readonly modifier when true', () => {
      wrapper.setProps({ readonly: true })
      expect(wrapper.find('.text-field--readonly').length).toBe(1)
    })
    it('Should not set the readonly modifier when false', () => {
      wrapper.setProps({ readonly: false })
      expect(wrapper.find('.text-field--readonly').length).toBe(0)
    })
  })
})
