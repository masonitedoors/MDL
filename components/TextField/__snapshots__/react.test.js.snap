// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render without crashing 1`] = `
<TextField
  align="left"
  disabled={false}
  error={false}
  helper=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  readonly={false}
  removeMargin={false}
  type="text"
  value=""
  variant="dark"
>
  <div
    className="text-field text-field--dark"
  >
    <div
      className="text-field__label"
    />
    <div
      className="text-field__helper"
    />
    <input
      className="text-field__input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      type="text"
      value=""
    />
    <TrailingIcon>
      <div
        className="text-field__icon"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
      />
    </TrailingIcon>
  </div>
</TextField>
`;
